### PR TITLE
KOGITO-178 - Index User Task data

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

File: data-index/data-index-service/src/main/resources/schema.graphqls [hunk 1]
```diff
@@ -56,3 +58,50 @@ input ProcessInstanceFilter {
...
+  potentialGroups: [String!]
+  potentialUsers: [String!]
+}
+
+input UserTaskInstanceFilter {
+  state: [String!]
+  id: [String!]
+  processInstanceId: [String!]
+  limit: Int
+  offset: Int
```
Comment (mswiderski) [inline data-index/data-index-service/src/main/resources/schema.graphqls]: can we have filter on owner, potential users/groups?

File: data-index/data-index-service/src/test/java/org/kie/kogito/index/IndexingServiceTest.java [hunk 2]
```diff
@@ -265,6 +345,7 @@ private String getProtoBufferFileV2() {
                 "   optional string name = 2;\n" +
                 "   optional string company = 3;\n" +
                 "   repeated org.kie.kogito.index.model.ProcessInstanceMeta processInstances = 4;\n" +
+                "   repeated org.kie.kogito.index.model.UserTaskInstanceMeta userTasks = 5;\n" +
```
Comment (mswiderski) [inline data-index/data-index-service/src/test/java/org/kie/kogito/index/IndexingServiceTest.java]: do we need to add this in the runtime generated Proto files?

Comment (cristianonicolai) [reply data-index/data-index-service/src/test/java/org/kie/kogito/index/IndexingServiceTest.java]: yep, this needs to be added there.

File: data-index/data-index-service/src/main/resources/schema.graphqls (hunk 1 above)
Comment (cristianonicolai) [reply data-index/data-index-service/src/main/resources/schema.graphqls]: sure, will add

File: data-index/data-index-service/src/main/resources/application.properties [hunk 3]
```diff
@@ -16,10 +16,20 @@ quarkus.infinispan-client.server-list=localhost:11222
...
 
 mp.messaging.incoming.kogito-processdomain-events.connector=smallrye-kafka
 mp.messaging.incoming.kogito-processdomain-events.topic=kogito-processinstances-events
-mp.messaging.incoming.kogito-processdomain-events.value.deserializer=org.kie.kogito.index.messaging.KogitoCloudEventDeserializer
-mp.messaging.incoming.kogito-processdomain-events.bootstrap.servers=localhost:9092
\ No newline at end of file
+mp.messaging.incoming.kogito-processdomain-events.value.deserializer=org.kie.kogito.index.messaging.KogitoProcessCloudEventDeserializer
+mp.messaging.incoming.kogito-processdomain-events.bootstrap.servers=localhost:9092
+
+mp.messaging.incoming.kogito-usertaskinstances-events.connector=smallrye-kafka
```
Comment (cristianonicolai) [inline data-index/data-index-service/src/main/resources/application.properties]: @ricardozanini fyi this brings in new topics to be setup

**Comment by cristianonicolai** [comment]
@mswiderski PR updated, please give it another go.

File: data-index/data-index-service/src/main/resources/application.properties (hunk 3 above)
Comment (ricardozanini) [reply data-index/data-index-service/src/main/resources/application.properties]: Updated: https://issues.jboss.org/browse/KOGITO-94

**Pull request merged**